### ISSUE TITLE
stany magazynowe

### ISSUE THREAD

**Issue by GregTora** [allegro/allegro-api#9558]
Mam dwa pytania:
1) Czy można w jakiś sposób "zaimportować" dane do oferty Allegro, tak aby oferta pokazywała aktualny stan magazynowy?
2) Jak można połączyć Allegro z ERP lub CMS, aby "na żywo" Allegro pokazywało aktualny stan magazynowy?

**Comment by PrzemyslawLukanowski** [comment]
Dane na temat aktualnego stanu magazynowego znajdziesz w odpowiedzi [GET /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation#operation/getProductOffer) w polu `stock.available`. Odpytując ten zasób, możesz pobierać dane o aktualnym stanie magazynowym w ofercie. Jeśli natomiast chcesz śledzić zmiany w tym polu, musisz skorzystać z dziennika ofertowego [GET /sale/offer-events](https://developer.allegro.pl/documentation#operation/getOfferEvents) - w przypadku zmiany liczby sztuk zwrócimy event `"OFFER_STOCK_CHANGED" `. Będzie to dla Ciebie informacja, że stock w ofercie uległ zmianie, zatem możesz pobrać dane oferty za pomocą GET /sale/product-offers/{offerId}, aby otrzymać aktualne dane oferty. 

Jeśli chcesz zmienić liczbę sztuk w ofercie, możesz skorzystać z [PATCH /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation#operation/editProductOffers) albo [PUT /sale/offer-quantity-change-commands/{commandId}](https://developer.allegro.pl/documentation#operation/quantityModificationCommandUsingPUT).

**Comment by GregTora** [comment]
Czy do tego celu trzeba utworzyć "nową aplikację w Allegro" i tam zdefiniować jakie dane mają być przekazywane?

**Comment by PrzemyslawLukanowski** [comment]
Tak, żeby wykonywać jakiekolwiek operacje w ramach naszego API, musisz posiadać zarejestrowaną aplikację, następnie zautoryzowaną przez użytkownika. Jeśli chodzi o to, jakie dane mają być przekazywane, to już zależy od Ciebie, z jakich konkretnie endpointów aplikacja będzie korzystać i na tej podstawie musisz nadać aplikacji odpowiednie [zakresy](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#scope-w-api-allegro).

**Comment by GregTora** [comment]
Jeżeli obecnie już mamy obsługiwaną aplikację, to czy do w/w celów mamy "utworzyć" nową aplikację?

**Comment by PrzemyslawLukanowski** [comment]
Nie musicie, wszelkie zmiany możecie wprowadzić w obecnej aplikacji.

**Comment by GregTora** [comment]
A jeżeli inna aplikacja jest używana do innych celów z inną firmą, a teraz potrzebujemy na inne cele z inną firmą to wówczas możemy utworzyć nową aplikację, czy jest jakiś limit aplikacji dla danego konta Allegro?

**Comment by AureliuszBrussy** [comment]
Na jednym koncie Allegro możesz [zarejestrować](https://apps.developer.allegro.pl/) 5 kluczy aplikacji. Natomiast zwróć uwagę, że jako sprzedający na Allegro możesz powiązać swoje konto z wieloma aplikacjami (czyli wyrazić zgodę na ich działanie w twoim imieniu).
